addons/cppcheckdata.py

```diff
@@ -750,36 +753,34 @@ class CppcheckData:
     def __init__(self, filename):
         self.configurations = []
 
-        data = ElementTree.parse(filename)
-
-        for platformNode in data.getroot():
-            if platformNode.tag == 'platform':
-                self.platform = Platform(platformNode)
-
-        for rawTokensNode in data.getroot():
-            if rawTokensNode.tag != 'rawtokens':
+        # Parse Cppcheck dump file incrementaly to avoid large memory consumption.
+        # Calling .clear() is necessary to let the element be garbage collected.
+        for _, node in ElementTree.iterparse(filename, events=('end',)):
+            if node.tag == 'dump':
+                self.configurations.append(Configuration(node))
+            elif node.tag == 'platform':
+                self.platform = Platform(node)
+            elif node.tag == 'suppressions':
+                for suppression_node in node:
+                    self.suppressions.append(Suppression(suppression_node))
+                    suppression_node.clear()
+            elif node.tag == 'rawtokens':
+                files = []
+                for rawtokens_node in node:
+                    if rawtokens_node.tag == 'file':
+                        files.append(rawtokens_node.get('name'))
+                    elif rawtokens_node.tag == 'tok':
+                        tok = Token(rawtokens_node)
+                        tok.file = files[int(rawtokens_node.get('fileIndex'))]
+                        self.rawTokens.append(tok)
```

Comment:
I do not understand the iterative parsing fully yet. Why is the `rawtokens_node` not cleared here at the end of every loop cycle? Is it still needed?

Reply:
I'm missing this change in git stage. When I added this, I've found, that maximum memory consumption went down about 0.2 MB.